server/service/src/invoice/inbound_return/update_lines/generate.rs

```diff
@@ -0,0 +1,131 @@
+use repository::{
+    EqualFilter, InvoiceLineFilter, InvoiceLineRepository, RepositoryError, StorageConnection,
+};
+
+use crate::{
+    invoice::InboundReturnLineInput,
+    invoice_line::{
+        stock_in_line::{DeleteStockInLine, InsertStockInLine, StockInType, UpdateStockInLine},
+        update_return_reason_id::UpdateLineReturnReason,
+    },
+};
+
+use super::UpdateInboundReturnLines;
+
+pub struct GenerateResult {
+    pub lines_to_add: Vec<InsertStockInLine>,
+    pub lines_to_update: Vec<UpdateStockInLine>,
+    pub lines_to_delete: Vec<DeleteStockInLine>,
+    pub update_line_return_reasons: Vec<UpdateLineReturnReason>,
+}
+
+pub fn generate(
+    connection: &StorageConnection,
+    UpdateInboundReturnLines {
+        inbound_return_id,
+        inbound_return_lines,
+    }: UpdateInboundReturnLines,
+) -> Result<GenerateResult, RepositoryError> {
+    let existing_lines = InvoiceLineRepository::new(connection).query_by_filter(
+        InvoiceLineFilter::new().invoice_id(EqualFilter::equal_to(&inbound_return_id)),
+    )?;
+    let check_already_exists = |id: &str| {
+        existing_lines
+            .iter()
+            .find(|line| line.invoice_line_row.id == id)
+            .is_some()
+    };
+
+    let lines_to_add = inbound_return_lines
```

Comment:
would it be weird to only iterate once and return a tuple? probably not an issue since there won't be a lot of lines, it just feels repetitive to loop over the same set of data four times.